### DISCUSSION TITLE
the diagnostic passcode (i.e. diag.passcode) seems not work

### DISCUSSION THREAD

**Discussion by ah-ping-luk** [belaban/JGroups#824] [General]
Hi,

  I have a TUNNEL transport and i want to enable disgnostic with passcode authorization. (tested on jgroups 5.2.23)

`e.g. <TUNNEL diag.enabled="true" diag.enable_tcp="true" diag.enable_udp="false" diag.bind_addr="0.0.0.0" diag.passcode="12345" />`

It seems the passcode setting do not work (no logs in server or Probe.sh)

Steps to reproduce:
1. start a jgroups app with TUNNEL + passcode
2. try to run the Probe.sh with proper passcode

**NOT work (with passcode)** : org.jgroups.tests.Probe -4 -udp false -tcp true -timeout 10000 **-passcode 12345** jmx=TUNNEL

**Work (no passcode)** : org.jgroups.tests.Probe -4 -udp false -tcp true -timeout 10000 jmx=TUNNEL
With result below:
```

TUNNEL={
....
diag.bind_addr=/0.0.0.0, diag.bind_interfaces=null, diag.enable_tcp=true, diag.enable_udp=false, diag.enabled=true, diag.mcast_addr=/224.0.75.75, diag.passcode=12345, diag.port=7500, diag.port_range=50, diag.running=true, diag.ttl=8
...
}

```
It seems the passcode is not correctly pickup in the TUNNEL?? It is strange that no exception throw in the TUNNEL class...

Is it that I miss some setting?

Thank you.

**Comment by belaban** [comment]
Oops, handling of `passcode` is only implemented for UDP, *not* TCP!

**Reply by ah-ping-luk** [reply]
Hi Bela, 

Ic...this information is missing in the document...!

Thanks for you help.

**Comment by belaban** [comment]
I'm working on fixing this: https://issues.redhat.com/browse/JGRP-2821, will be done today

**Comment by belaban** [comment]
I just released 5.3.11 and 5.2.28; the latter is the one you probably want to upgrade to
